driver/src/contracts/snapp_contract.rs

```diff
@@ -1,17 +1,34 @@
 #[cfg(test)]
 extern crate mock_it;
 
-use web3::contract::{Contract, Options};
+use web3::contract::Options;
 use web3::futures::Future;
-use web3::types::{Address, BlockId, H256, U128, U256};
+use web3::types::{BlockId, H256, U128, U256};
 
 use crate::error::DriverError;
 
+use super::base_contract::BaseContract;
+
 use std::env;
 use std::fs;
 
 type Result<T> = std::result::Result<T, DriverError>;
 
+pub struct SnappContractImpl {
+    base: BaseContract
+}
+
+impl SnappContractImpl {
+    pub fn new() -> Self {
+        let contract_json = fs::read_to_string("dex-contracts/build/contracts/SnappAuction.json").unwrap();
+        let address = env::var("SNAPP_CONTRACT_ADDRESS").unwrap();
+        SnappContractImpl {
+            base: BaseContract::new(address, contract_json).unwrap()
```

Comment:
Now, that this can fail, I'd suggest to return a Result<Self> from this method.

Reply:
That is cool, although I have since changed the implementation and moved all the environment variables into the new (as suggested).